zrml/prediction-markets/src/lib.rs

```diff
@@ -2236,27 +2332,42 @@ mod pallet {
             ) -> DispatchResult,
         {
             let dispute_period = T::DisputePeriod::get();
+
             if now <= dispute_period {
                 return Ok(());
             }
 
-            let block = now.saturating_sub(dispute_period);
+            let normal_dispute_start_block = now.saturating_sub(dispute_period);
 
             // Resolve all regularly reported markets.
-            for id in MarketIdsPerReportBlock::<T>::get(&block).iter() {
+            for id in MarketIdsPerReportBlock::<T>::get(&normal_dispute_start_block).iter() {
                 let market = T::MarketCommons::market(id)?;
                 if let MarketStatus::Reported = market.status {
                     cb(id, &market)?;
                 }
             }
-            MarketIdsPerReportBlock::<T>::remove(&block);
+            MarketIdsPerReportBlock::<T>::remove(&normal_dispute_start_block);
+
+            let mut resolve_disputed_markets = |start_block| -> DispatchResult {
+                for id in MarketIdsPerDisputeBlock::<T>::get(&start_block).iter() {
+                    let market = T::MarketCommons::market(id)?;
+                    cb(id, &market)?;
+                }
+                MarketIdsPerDisputeBlock::<T>::remove(&start_block);
+                Ok(())
+            };
 
             // Resolve any disputed markets.
-            for id in MarketIdsPerDisputeBlock::<T>::get(&block).iter() {
-                let market = T::MarketCommons::market(id)?;
-                cb(id, &market)?;
+            resolve_disputed_markets(normal_dispute_start_block)?;
+
+            let global_dispute_period = T::GlobalDisputePeriod::get();
+            if now <= global_dispute_period {
+                return Ok(());
             }
-            MarketIdsPerDisputeBlock::<T>::remove(&block);
+            let global_dispute_start_block = now.saturating_sub(global_dispute_period);
+
+            // Resolve any global dispute markets.
+            resolve_disputed_markets(global_dispute_start_block)?;
```

Comment:
This does not make sense. If `GlobalDisputePeriod => DisputePeriod`, then all markets will be resolved using the normal dispute scheme. If `GlobalDisputePeriod < DisputePeriod`, all markets will be resolved using the global dispute scheme.

Reply:
Yes you are right. This is a head struggle. I looked at the PR from @vivekvpandya . There you can easily put in the future block number. That's the approach I'd like to follow now.

https://github.com/zeitgeistpm/zeitgeist/blob/65b64b52b9418dc48db01ea61f96b12e68291df8/zrml/prediction-markets/src/lib.rs#L427-L431

Reply:
080a8160f9be64c844f0ae197e8dcd502cd558b4